File: src/main/java/com/iminling/core/annotation/Authentication.java
package com.iminling.core.annotation;

import java.lang.annotation.*;

@Target({ElementType.METHOD, ElementType.TYPE})
@Retention(RetentionPolicy.RUNTIME)
@Documented
@LoginRequired
public @interface Authentication {

    String[] roles() default {};

    String[] permissions() default {};

}

File: iminling-core/src/main/kotlin/com/iminling/core/annotation/Authentication.kt
package com.iminling.core.annotation

/**
 * @author  yslao@outlook.com
 * @since  2021/11/28
 */
@Target(AnnotationTarget.FUNCTION, AnnotationTarget.TYPE)
@Retention(AnnotationRetention.RUNTIME)
@MustBeDocumented
@LoginRequired
annotation class Authentication(

    /**
     * 角色
     */
    val roles: Array<String>,

    /**
     * 权限
     */
    val permissions: Array<String>
)
